pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentOpenHashSet.java

```diff
@@ -598,8 +598,14 @@ static final <K> long hash(K key) {
         return hash;
     }
 
-    static final int signSafeMod(long n, int max) {
-        return (int) n & (max - 1);
+    static final int signSafeMod(long dividend, int divisor) {
+        int mod = (int) (dividend % divisor);
+
+        if (mod < 0) {
+            mod += divisor;
+        }
+
+        return mod;
```

Comment:
Can we move to a Utility Class for computing the sign safe mod?
We have a similar [Unitity Class](https://github.com/apache/pulsar/blob/c675a3dd6b4a3a6009d2a10c38ee79d8219b1920/pulsar-client/src/main/java/org/apache/pulsar/client/util/MathUtils.java#L24) in pulsar-client, not pulsar-client-api.
Is it better to move it to pulsar-common to reduce the duplicated code?